polyfill/lib/date.mjs

```diff
@@ -146,13 +137,13 @@ export class Date {
     const MonthDay = ES.GetIntrinsic('%Temporal.MonthDay%');
     return new MonthDay(GetSlot(this, MONTH), GetSlot(this, DAY));
   }
-  static from(arg) {
-    let result = ES.ToDate(arg);
+  static from(arg, { disambiguation = 'constrain' } = {}) {
+    disambiguation = ES.ToString(disambiguation);
```

Comment:
Should check if `disambiguation` is one of the allowed values. (`ToDate` misses this check at least in the case that `arg` is a `Date` already.) Test.